src/mdtest.c

```diff
@@ -1021,25 +1026,77 @@ int updateStoneWallIterations(int iteration, uint64_t items_done, double tstart,
 
   VERBOSE(1,1,"stonewall hit with %lld items", (long long) items_done );
   MPI_Allreduce(& items_done, & max_iter, 1, MPI_LONG_LONG_INT, MPI_MAX, testComm);
-  summary_table[iteration].stonewall_time[MDTEST_FILE_CREATE_NUM] = GetTimeStamp() - tstart;
+  o.summary_table[iteration].stonewall_time[MDTEST_FILE_CREATE_NUM] = GetTimeStamp() - tstart;
   *out_max_iter = max_iter;
 
   // continue to the maximum...
   long long min_accessed = 0;
   MPI_Reduce(& items_done, & min_accessed, 1, MPI_LONG_LONG_INT, MPI_MIN, 0, testComm);
   long long sum_accessed = 0;
   MPI_Reduce(& items_done, & sum_accessed, 1, MPI_LONG_LONG_INT, MPI_SUM, 0, testComm);
-  summary_table[iteration].stonewall_item_sum[MDTEST_FILE_CREATE_NUM] = sum_accessed;
-  summary_table[iteration].stonewall_item_min[MDTEST_FILE_CREATE_NUM] = min_accessed * size;
+  o.summary_table[iteration].stonewall_item_sum[MDTEST_FILE_CREATE_NUM] = sum_accessed;
+  o.summary_table[iteration].stonewall_item_min[MDTEST_FILE_CREATE_NUM] = min_accessed * o.size;
 
-  if(items != (sum_accessed / size)){
-    VERBOSE(0,-1, "Continue stonewall hit min: %lld max: %lld avg: %.1f \n", min_accessed, max_iter, ((double) sum_accessed) / size);
+  if(o.items != (sum_accessed / o.size)){
+    VERBOSE(0,-1, "Continue stonewall hit min: %lld max: %lld avg: %.1f \n", min_accessed, max_iter, ((double) sum_accessed) / o.size);
     hit = 1;
   }
 
   return hit;
 }
 
+void file_test_create(const int iteration, const int ntasks, const char *path, rank_progress_t * progress, double *t){
```

Comment:
It would be better to move this code to its own function in a different patch.  This patch is already changing several hundred lines to use the options structure, and adding other changes into the patch makes it harder to review.

Reply:
I agree, will do it next time. It was part of the attempt to refactor the "stonewall" which then lead to the situation that I refactored the "static" pieces instead to increase understandability.